lib/run_loop/fifo.rb

```diff
@@ -22,6 +22,9 @@ def self.write(pipe, msg, options={})
               wrote = pipe_io.write_nonblock msg
               bytes_written += wrote
               msg = msg[wrote..-1]
+            rescue Errno::EPIPE
```

Comment:
Hey @TeresaP 

I am intrigued.  I am concerned about the use of `flush` here and an unguarded `retry` which might cause an infinite loop.

The Errno::EPIPE is raised when there is an interruption in the the _non blocking_ write that is being performed.

In this case, run-loop is trying to write something to the REPL, but UIAutomation (or instruments) interrupts it.

Have you tried this:

```
 # Rescue Erro::EPIPE with the other errors
 rescue IO::WaitWritable, Errno::EINTR, Errno::EPIPE
```

The `flush` is problematic because we don't want to force the write because there is new information in the REPL - we might perform an incomplete write.

I _believe_ the IO.select will correctly continue the write - there is a  good chance that I am wrong about this.

Can you try two things:

1. Update to ruby >= 2.2; preferably ruby 2.3.0 and see if you can reproduce without this change.
2. Try the change I suggested with any ruby version.

Based on a recent conversation, I believe you have experience with rbenv.  Install it again and use the .ruby-version facility to pin the version of ruby for your project.

```
# Forces ruby 2.3.0 in this directory and all subdirectories.
$ rbenv local 2.3.0

# Forces ruby 2.2.0 in this directory and all subdirectories.
$ rbenv local 2.2.0
```

`rbenv local` creates a .ruby-version file; commit that to your VCS so the ruby version will be pinned across your team.

I use the following script to install rubies with rbenv.  It assumes you have installed:

* openssl
* readline

with homebrew.

```
#!/usr/bin/env bash
RUBY_CONFIGURE_OPTS="--with-readline-dir=$(brew --prefix readline) --with-openssl-dir=$(brew --prefix openssl)" \
  rbenv install "${1}"
```

Do you have an app and some tests that reproduce this always?  If so, can we pair to see if we can work out what is happening?